contracts/src/v0.8/functions/tests/v1_X/FunctionsCoordinator.t.sol

```diff
@@ -17,48 +21,220 @@ contract FunctionsCoordinator_Constructor is FunctionsRouterSetup {
-contract FunctionsCoordinator_GetDONPublicKey {
+contract FunctionsCoordinator_GetDONPublicKey is FunctionsDONSetup {
+  function test_GetDONPublicKey_RevertIfEmpty() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    // Reverts when empty
+    vm.expectRevert(FunctionsCoordinator.EmptyPublicKey.selector);
+    s_functionsCoordinator.getDONPublicKey();
+  }
+
+  function test_GetDONPublicKey_Success() public {
+    s_functionsCoordinator.setDONPublicKey(s_donKey);
+
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
 
+    bytes memory donKey = s_functionsCoordinator.getDONPublicKey();
+    assertEq(donKey, s_donKey);
+  }
 }
 
 /// @notice #setDONPublicKey
-contract FunctionsCoordinator__SetDONPublicKey {
+contract FunctionsCoordinator_SetDONPublicKey is FunctionsDONSetup {
+  function test_SetDONPublicKey_RevertNotOwner() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    vm.expectRevert("Only callable by owner");
+    s_functionsCoordinator.setDONPublicKey(s_donKey);
+  }
+
+  function test_SetDONPublicKey_Success() public {
+    s_functionsCoordinator.setDONPublicKey(s_donKey);
 
+    bytes memory donKey = s_functionsCoordinator.getDONPublicKey();
+    assertEq(donKey, s_donKey);
+  }
 }
 
 /// @notice #_isTransmitter
-contract FunctionsCoordinator_IsTransmitter {
-  // TODO: make contract internal function helper
+contract FunctionsCoordinator__IsTransmitter is FunctionsDONSetup {
+  function test__IsTransmitter_SuccessFound() public {
+    bool isTransmitter = s_functionsCoordinator.isTransmitter_HARNESS(NOP_TRANSMITTER_ADDRESS_1);
+    assertEq(isTransmitter, true);
+  }
+
+  function test__IsTransmitter_SuccessNotFound() public {
+    bool isTransmitter = s_functionsCoordinator.isTransmitter_HARNESS(STRANGER_ADDRESS);
+    assertEq(isTransmitter, false);
+  }
 }
 
-/// @notice #setNodePublicKey
-contract FunctionsCoordinator_SetNodePublicKey {
+/// @notice #startRequest
+contract FunctionsCoordinator_StartRequest is FunctionsSubscriptionSetup {
+  function test_StartRequest_RevertIfNotRouter() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
 
-}
+    vm.expectRevert(Routable.OnlyCallableByRouter.selector);
 
-/// @notice #deleteNodePublicKey
-contract FunctionsCoordinator_DeleteNodePublicKey {
+    s_functionsCoordinator.startRequest(
+      FunctionsResponse.RequestMeta({
+        requestingContract: address(s_functionsClient),
+        data: new bytes(0),
+        subscriptionId: s_subscriptionId,
+        dataVersion: FunctionsRequest.REQUEST_DATA_VERSION,
+        flags: bytes32(0),
+        callbackGasLimit: 5_500,
+        adminFee: s_adminFee,
+        initiatedRequests: 0,
+        completedRequests: 0,
+        availableBalance: s_subscriptionInitialFunding,
+        subscriptionOwner: OWNER_ADDRESS
+      })
+    );
+  }
 
-}
+  event OracleRequest(
```

Comment:
Why do we need to copy-paste this definition instead of including the real one?

Reply:
It's just how solidity & foundry work. Events aren't exposed from the contract interface.
(see https://ethereum.stackexchange.com/questions/141242/to-run-vm-expectemit-in-my-foundry-test-do-i-need-to-put-the-events-in-the-te , we could make a specific event interface, but not sure I find that any different)